### ISSUE TITLE
git-pulls won't freshen all repos

### ISSUE THREAD

**Issue by pfuntner** [pfuntner/toys#96]
I made some non-trivial improvements to git-pulls recently and I think I broke something.  Here's what I've observed:

1. Use `git-pulls`, defaulting to `~/repos`
2. Observe that one repo was freshened
3. There was still at least one repo that was not freshened!

I just ran the script a few minutes ago on my home computer and observed only `~/repos/fun` freshened - `~/repos/toys` was not and `toys` is much more important to m than `fun`.  I don't know what's wrong but I need to look into it ASAP.

**Comment by pfuntner** [comment]
I tried to recreate this and did the following:
1. Created two small new repos with just a readme file
2. Clone the repos into a new directory on my local computer
3. Updated the readme file on both repos
4. Ran `git-pulls` on the directory holding the two repos
5. Observed that both repos were updated

Whatever's wrong, the problem did not manifest.

**Comment by pfuntner** [comment]
I tried a slightly different scenario and did:
1. Created a third repo and cloned it
2. I updated the readme of the first and third-repos
3. Ran `git-pulls` again
4. Observed the repos were freshened - they were done alphabetically and the second one was skipped:
    ```
    Path     Branch  Old SHA1  Old Date             Old Text          New SHA1  New Date             New Text
    ./test1  main    6ea7cd7   2022-07-09 08:14:17  Update README.md  5cba679   2022-07-09 08:20:08  Update README.md
    ./test2  main    57599b2   2022-07-09 08:14:30  Update README.md
    ./test3  main    942d87c   2022-07-09 08:19:19  Create README.md  5cef327   2022-07-09 08:19:57  Update README.md
    ```

**Comment by pfuntner** [comment]
I looked at the code and don't see any obvious mistakes but I made a couple of changes in 3b07744:

- I got rid of a useless option I had added to the `run()` method
- I added a debugging message to show the result of looking to see if there are updates during a pull

**Comment by pfuntner** [comment]
I came up with another way to attack this and did the following:

1. Cloned the same repos in `~/repos` in another directory - brand new up-to-date clones
2. Used `git rebase -i` to remove commits from the last couple of days from the `toys` and `fun` repos
3. Tried doing `git-pulls --dry-run .` and it failed so I'll investigate that:
    ```
    $ git-pulls --dry-run .
    2022-07-09 09:06:20,444 WARNING /home/mrbruno/bin/git-pulls:130 Note: local repos are not being changed cause of --dry-run
    Traceback (most recent call last):
      File "/home/mrbruno/bin/git-pulls", line 147, in <module>
        (date, sha1, text) = get_last_commit()
      File "/home/mrbruno/bin/git-pulls", line 77, in get_last_commit
        return (date, sha1[:7], text)
    TypeError: 'NoneType' object is not subscriptable
    $
    ```

**Comment by pfuntner** [comment]
One of the first things I did Monday morning after the changes I made from home, I refreshed my repos on my work laptop and everyday Ubuntu machine.  They seemed to work as expected.  I may get into the habit of doing:

```
$ git-pulls -vv
$ git-pulls --dry-run
# verify that no repos need to be freshened
```

SInce `--dry-run` takes a very different code path, I think that's a good check.

**Comment by pfuntner** [comment]
A week later, I tried again after making updates to the repos at work:
1. `git pulls` did nothing
2. I ran `git-pulls -vv` and captured all the output: [git-log output.txt](https://github.com/pfuntner/toys/files/9225918/git-log.output.txt)
3. I `cd`ed into `bin` and did a direct `git pull` and the changes were pulled fine

**Comment by pfuntner** [comment]
I also recreated the issue:

1. Copied all my repos to a temporary directory
2. Observed the top commits of the toys repo:
    ```
    Commit                                    Author                              Date (UTC)           Text
    504459bdf5ff1e7aaf8b82078e6475b1dd573a5e  jpfuntne@cisco.com                  2022-07-29 17:14:33  Support --executable option for podman, nerdctl, etc
    39c446f1573ba3b3ea927a75e73babab57fb62f7  jpfuntne@cisco.com                  2022-07-25 12:20:04  Fixing typo
    f152d1a9937c09692cedc064593f4a695d458a77  jpfuntne@cisco.com                  2022-07-25 11:41:13  Use timestamp from old file
    4e9ef34d7e5adc1ea90597eba6912b5287cf0652  jpfuntne@cisco.com                  2022-07-14 14:56:38  Adding bash sebang so xargs can run the script
    571a85341f5dfa70534e80c95200cb629f893d30  jpfuntne@cisco.com                  2022-07-11 13:10:21  Improvement to handle lists\n\nI ran into a scenario where I
    bc4f3f513502c9e23fe910037f40906edea8fc69  pfuntner@pobox.com                  2022-07-09 13:37:39  Fixing some "dry run" issues
    3b0774425c3425a1af68882db2280a0c5a3bcccb  pfuntner@pobox.com                  2022-07-09 12:58:41  Adding a debugging message to help chase issue #96
    1da21158d7e3331b2a500f4999021a8af4852cf4  jpfuntne@cisco.com                  2022-07-08 16:49:37  Commiting some tools I never got around to pushing up
    0869557fc49e5ad602de7dae3e08ff9ef10eb368  jpfuntne@cisco.com                  2022-07-08 16:46:39  Adding support for multiple repos\n\nI had the need to scan
    ```
3. Used `git rebase` to remove the top three commits locally - those were created during the week
4. Observed the top commits of the toys repo after removing commits:
    ```
    Commit                                    Author                              Date (UTC)           Text
    4e9ef34d7e5adc1ea90597eba6912b5287cf0652  jpfuntne@cisco.com                  2022-07-14 14:56:38  Adding bash sebang so xargs can run the script
    571a85341f5dfa70534e80c95200cb629f893d30  jpfuntne@cisco.com                  2022-07-11 13:10:21  Improvement to handle lists\n\nI ran into a scenario where I
    bc3f3f513502c9e23fe910037f40906edea8fc69  pfuntner@pobox.com                  2022-07-09 13:37:39  Fixing some "dry run" issues
    3b0774425c3425a1af68882db2280a0c5a3bcccb  pfuntner@pobox.com                  2022-07-09 12:58:41  Adding a debugging message to help chase issue #96
    1da21158d7e3331b2a500f4999021a8af4852cf4  jpfuntne@cisco.com                  2022-07-08 16:49:37  Commiting some tools I never got around to pushing up
    0869557fc49e5ad602de7dae3e08ff9ef10eb368  jpfuntne@cisco.com                  2022-07-08 16:46:39  Adding support for multiple repos\n\nI had the need to scan
    4616783f6311cb5b84d78b554c03ed8b6a91840d  jpfuntne@cisco.com                  2022-07-07 12:57:07  creating new tool
    42eeeb58e8dd8127c50c84386ab6edd6f7c3c291  jpfuntne@cisco.com                  2022-06-24 13:30:52  Creating new tool
    a34d4937622a7c69064511d6d54bd773e385e0e7  jpfuntne@cisco.com                  2022-06-24 11:48:18  Elaborating on script purpose
    ```
5. Tried to pull and observed nothing was pulled:
    ```
    [mrbruno@bruno-meerkat repos]$ git-pulls .
    Path                  Branch  Old SHA1  Old Date             Old Text                New SHA1  New Date  New Text
    ./bruno-ansible       master  b1452a9   2022-05-17 06:54:19  Creating playbook to
    ./clamav-onacc        main    2832ce4   2021-03-12 15:21:09  Latest changes\n\nMore
    ./docker              master  6f4160e   2019-09-13 13:42:54  I got the makefile t
    ./fun                 master  6ec983f   2022-07-13 08:56:19  Moving shlex import
    ./iyt                 master  6f58114   2019-07-06 11:51:38  Creating gather.sh s
    ./only-bruno          main    5a87a92   2021-12-13 18:55:32  fixing typo
    ./pfuntner.github.io  master  40d3412   2020-11-07 09:05:29  Creating separate st
    ./vagrant             master  8bb1165   2019-06-11 08:00:04  Automating --reload
    [mrbruno@bruno-meerkat toys master]$ 
    ```

**Comment by pfuntner** [comment]
I'm only just now realizing that `toys` wasn't in the list of repos!

**Comment by pfuntner** [comment]
The problem is because on my home computer, `~/repos/toys` is a symlink into my extra disk - the others are not.  The script has logic to skip symlinks because at work I have some repos nested other another via symlinks.  I think a better solution is to just case the hunt after a `.git` directory is found.

**Comment by pfuntner** [comment]
Fixed with 700dc25
